### ISSUE TITLE
Incorrect Variable Bindings in Postgres Transactions

### ISSUE THREAD

**Issue by loopnyapy** [simolus3/moor#2894]
**What we are trying to do**
We are trying to run queries within a transaction that includes `db.insertReturning`.

**Expected**
I expect that, similar to any other query not within a transaction, `?` should be replaced with `@`, and the appropriate method, such as `PgDelegate.runInsert`, should be called for `db.insertReturning`.

**Actual**
Instead, `PgDelegate.runSelect` is called with an unprepared `statement` argument, resulting in the use of variable binding `?` instead of `@`.

Simple code example:
```dart
  Future<Tag> createTag(Tag tag) =>
      database.transaction(() {
        return database
            .into(database.tags)
            .insertReturning(tag.toCompanion());
      });
```

Pubspec.yaml:
```yaml
dependencies:
  drift_postgres: ^1.1.0
  drift: ^2.15.0

dev_dependencies:
  drift_dev: ^2.15.0
```

I look forward to any insights or solutions regarding this issue. Thank you 🙏

**Comment by simolus3** [comment]
Thanks for the report. It's possible that there's a bug somewhere making the query builder believe it's targeting sqlite3, but unfortunately I couldn't reproduce it.

I'm using the example in `extras/drift_postgres/example` with this patch to run a returning insert in a transaction:

```patch
diff --git a/extras/drift_postgres/example/main.dart b/extras/drift_postgres/example/main.dart
index ad85ad51..3da6b050 100644
--- a/extras/drift_postgres/example/main.dart
+++ b/extras/drift_postgres/example/main.dart
@@ -37,11 +37,14 @@ void main() async {
     logStatements: true,
   ));
 
-  final user = await database.users.insertReturning(
-      UsersCompanion.insert(name: 'Simon', id: Value(Uuid().v4obj())));
-  print(user);
+  await database.transaction(() async {
+    final user = await database.into(database.users).insertReturning(
+        UsersCompanion.insert(name: 'Simon', id: Value(Uuid().v4obj())));
 
-  await database.users.deleteOne(user);
+    print(user);
+
+    await database.users.deleteOne(user);
+  });
 
   await database.close();
 }
```

However, that still makes it use `$` as expected.

**Comment by loopnyapy** [comment]
Thank you! Thats right, `insertReturning` works within a transaction just fine. However, I forgot to mention, in our case we are using `DriftIsolate`, and this leads to unexpected behaviour as I described.

Here is the repository with code to reproduce this bug.
https://github.com/loopnyapy/drift-tx-bug

**Comment by simolus3** [comment]
Thanks - I can reproduce the issue with drift 2.15.0, but it seems to be fixed on `develop` already. I don't remember fixing this particular problem though :sweat_smile: I'll write a test for this to make sure it doesn't break again - then I'm hoping to release the new version towards the end of the week.

**Comment by loopnyapy** [comment]
Awesome! Thanks a lot 🙏

**Comment by simolus3** [comment]
Should be fixed in one of the versions released since the linked commit (so 2.16+ I think).
